server/TypeDBServer.java

```diff
@@ -89,7 +90,9 @@ protected TypeDBServer(Configuration configuration, boolean debug, Factory facto
 
         Options.Database options = new Options.Database()
                 .typeDBDir(getTypedbDir())
-                .dataDir(configuration.dataDir())
+                .dataDir(configuration.storage().dataDir())
+                .storageDataCacheSize(configuration.storage().dbCache().dataSize())
+                .storageIndexCacheSize(configuration.storage().dbCache().indexSize())
```

Comment:
using the updated YAML configuration that now has a `Storage` sub-section for configuring storage options